ir/src/main/scala/org/scalajs/core/ir/Types.scala

```diff
@@ -184,7 +184,11 @@ object Types {
             if (isPrimitiveClass(lhsBase) || isPrimitiveClass(rhsBase)) {
               lhsBase == rhsBase
             } else {
-              isSubclass(lhsBase, rhsBase)
+              /* All things must be considered subclasses of Object for this
+               * purpose, even raw JS types and interfaces that do not have
+               * Object in their ancestors, strictly speaking.
```

Comment:
Hum, nothing, really. I guess when I wrote that I was still surprised myself to rediscover that interfaces did not have `Object` in their ancestors.